src/redux/authSlice.ts

```diff
@@ -258,37 +263,45 @@ export const authSlice = createSlice({
 
                 setLSNotSignedIn();
             })
-            .addCase(fetchResourcePermissions.pending, (state, { meta }) => {
-                const key = makeResourceKey(meta.arg.resource);
-                state.resourcePermissions[key] = {
-                    ...state.resourcePermissions[key],
-                    isFetching: true,
-                    hasAttempted: false,
-                    permissions: [],
-                    error: "",
-                };
+            .addCase(fetchResourcesPermissions.pending, (state, { meta }) => {
+                for (const resource of meta.arg.resources) {
+                    const key = makeResourceKey(resource);
+                    state.resourcePermissions[key] = {
+                        ...state.resourcePermissions[key],
+                        isFetching: true,
+                        hasAttempted: false,
+                        permissions: [],
+                        error: "",
+                    };
+                }
             })
-            .addCase(fetchResourcePermissions.fulfilled, (state, { meta, payload }) => {
-                const key = makeResourceKey(meta.arg.resource);
-                state.resourcePermissions[key] = {
-                    ...state.resourcePermissions[key],
-                    isFetching: false,
-                    hasAttempted: true,
-                    permissions: payload?.result?.[0] ?? [],
-                };
+            .addCase(fetchResourcesPermissions.fulfilled, (state, { meta, payload }) => {
+                const resources = meta.arg.resources;
+                for (const r in resources) {
+                    const key = makeResourceKey(resources[r]);
```

Comment:
Make uniform with the other loops

Reply:
like with the resources var? in this case its done differently because we index into the payload + access `resources` more than once